Objects/exceptions.c

```diff
@@ -2708,6 +2708,68 @@ set_unicodefromstring(PyObject **attr, const char *value)
     return 0;
 }
 
+/*
+ * Adjust the (inclusive) 'start' value of a UnicodeError object.
+ *
+ * The START can be negative or not, but when adjusting the value,
+ * we clip it in [0, MAX(0, OBJLEN - 1)] but do not intepret it as
+ * a relative offset.
+ */
+#define UNICODE_ERROR_ADJUST_START(START, OBJLEN)   \
+    do {                                            \
+        assert(OBJLEN >= 0);                        \
```

Comment:
Macro arguments should be in parentheses, e.g.:
```suggestion
        assert((OBJLEN) >= 0);                      \
```

But I don't see a reason to use a macro here; you can do:
```c
static inline Py_ssize_t
UnicodeError_adjusted_start(Py_ssize_t start, Py_ssize_t objlen)
{
    assert(objlen >= 0);
    if (start < 0) {
        return 0;
    }
    if (start >= objlen) {
        return objlen == 0 ? 0 : objlen - 1;
    }
    return start;
}
```

Reply:
> Macro arguments should be in parentheses, e.g.:

Thanks, I forgot.

> But I don't see a reason to use a macro here; you can do:

Habits. I'll convert it into a regular function (thanks).